byzcoin/bcadmin/main.go

```diff
@@ -295,9 +434,458 @@ func add(c *cli.Context) error {
 	return nil
 }
 
-func keys(c *cli.Context) error {
-	s := darc.NewSignerEd25519(nil, nil)
-	fmt.Println("Identity:", s.Identity())
-	fmt.Printf("export PRIVATE_KEY=%v\n", s.Ed25519.Secret)
+func key(c *cli.Context) error {
+	newSigner := darc.NewSignerEd25519(nil, nil)
+	lib.SaveKey(newSigner)
+
+	save := c.String("save")
+	if save == "" {
```

Comment:
Don't repeat what goes into the file twice, instead set fo correctly before and then call fmt.Println(fd,...) one time:
```go
var fo io.Writer
var err error
if save == "" {
  fo = os.Stdout
} else {
  fo, err = os.Create(save)
  // check err
  defer fo.Close()
}
fmt.Fprintln(fo, newSigner.Identity.String())
```

Another way to go about it would be to use some kind of writeToFileOrStdout(fn, w) wrapper which would take care of this for you all in one place. Then use fmt.Fprintln to write into a *bytes.Buffer (which is an io.ReadWriter) and then give the buf to your writeToFileOrStdout where it would use io.Copy to send the output to the right file descriptor.